src/lib/core/ErrorStr.cpp

```diff
@@ -50,8 +50,25 @@ static ErrorFormatter * sErrorFormatterList = nullptr;
  */
 DLL_EXPORT const char * ErrorStr(CHIP_ERROR err, bool withSourceLocation)
 {
-    char * formattedError   = sErrorStr;
-    uint16_t formattedSpace = sizeof(sErrorStr);
+    ErrorStr(err, withSourceLocation, sErrorStr, sizeof(sErrorStr));
+    return sErrorStr;
+}
+
+/**
+ * This routine writess a human-readable NULL-terminated C string into the buf
+ * which describes the provided error.
+ *
+ * @param[in] err                      The error for format and describe.
+ * @param[in] withSourceLocation       Whether or not to include the source
+ * @param[in] buf                      Provided buf to write into
+ * @param[in] bufSize                  Size of the buf
+ * location in the output string. Only used if CHIP_CONFIG_ERROR_SOURCE &&
+ * !CHIP_CONFIG_SHORT_ERROR_STR. Defaults to true.
+ */
+DLL_EXPORT void ErrorStr(CHIP_ERROR err, bool withSourceLocation, char * buf, uint16_t bufSize)
```

Comment:
Use a span for ptr + size

Reply:
Maybe even consider something like a dedicated storage type like https://github.com/project-chip/connectedhomeip/blob/master/src/app/data-model-provider/ActionReturnStatus.h#L100

Reply:
Using a span in ErrorStr.cpp seems inevitably introducing circular dependancy.
Span uses CodeUtils.h and CodeUtils.h uses Errors which I think is pretty hard to break up.

Reply:
Ah, didn't see your comment about the storage type, will try to see if that helps.